### PR TITLE
android: Add key warning

### PR DESCRIPTION
**Pull request by t895**
Now like the desktop app, you'll get a warning on startup if you don't have the keys required to decrypt games/firmware.
image

Additionally, this removes the "Reinitialize keys" option in the desktop app which has been useless since firmware 6.2.0

### CODE REVIEW

File: src/yuzu/main.cpp [hunk 1]
```diff
@@ -4551,85 +4548,9 @@ void GMainWindow::OnMouseActivity() {
...
-    bool all_keys_present{true};
-
-    if (keys.BaseDeriveNecessary()) {
-        Core::Crypto::PartitionDataManager pdm{vfs->OpenDirectory("", FileSys::Mode::Read)};
-
-        const auto function = [this, &keys, &pdm] {
-            keys.PopulateFromPartitionData(pdm);
-
-            system->GetFileSystemController().CreateFactories(*vfs);
-            keys.DeriveETicket(pdm, system->GetContentProvider());
```
Comment (liamwhite) [inline src/yuzu/main.cpp:4587]: ES key derivation needs to be preserved to allow dumps made with console-specific data to continue to operate

**Comment by liamwhite** [comment]
Another issue here is that it seems you will no longer get a warning on startup if you don't have firmware and also don't provide any keys, when you previously did.

**Comment by t895** [comment]
Got the popup back and fixed up the firmware check so it actually works for desktop now

**Comment by t895** [comment]
I just removed the firmware check and fixed up weird situations where the key check would fail during setup

**Review by liamwhite — APPROVED** [review]

**Pull request merged**